### PR TITLE
FIX: No JIT when quoting a mention

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/composer-editor.js [hunk 1]
```diff
@@ -860,6 +864,18 @@ export default Component.extend({
     this.send("togglePreview");
   },
 
+  _isInQuote(element) {
+    for (let e = element; e; e = e.parentElement) {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/composer-editor.js:868]: if I read this correctly you want to check `element` and `element.parentElement` right?

I think a more readable solution would be:

```javascript
[element, element.parentElement].forEach(e => {
  // code
});
```

Comment (ZogStriP) [reply app/assets/javascripts/discourse/app/components/composer-editor.js:868]: I think the for loop here go back up the whole tree. Not just one parent 😉

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/components/composer-editor.js:868]: yes indeed, hmmm well then ok 🤷‍♂️

Comment (AndrewPrigorshnev) [reply app/assets/javascripts/discourse/app/components/composer-editor.js:868]: Such loops implemented with `for` can be confusing. I have this version with `while` more self explainable:

```javascript
  _isInQuote(element) {
    let parent = element.parentElement;
    while (parent && !this._isPreviewRoot(parent)) {
      if (this._isQuote(parent)) {
        return true;
      }

      parent = parent.parentElement;
    }

    return false;
  },

  _isPreviewRoot(element) {
    return (
      element.tagName === "DIV" &&
      element.classList.contains("d-editor-preview")
    );
  },

  _isQuote(element) {
    return element.tagName === "ASIDE" && element.classList.contains("quote");
  },
```

**Review by jjaffeux — APPROVED** [review]

**Review by ZogStriP — APPROVED** [review]

**Pull request merged**